js/runtime.js

```diff
@@ -572,16 +566,40 @@
         },
 
         shape: {
-            fillShape: function(shp){
-                shp();
+            fill: function(pathSet){
```

Comment:
This comes with drawbacks, in that you have to define a shape exactly when you are drawing it, and that there might be problems if we make blocks that tells us things about shapes we dont want to draw. If I actually draw the paths in fill or stroke, though, a shape (or pathset) is really just a set of paths TO be drawn, not ones that already have been